src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
+        if (json.HasKey(L"zone-index-set"))
         {
-            auto history = appZoneHistoryMap.find(processPath);
-            if (history != std::end(appZoneHistoryMap))
+            data.zoneIndexSet = {};
+            for (auto& value : json.GetNamedArray(L"zone-index-set"))
             {
-                auto& perDesktopData = history->second;
-                for (auto data = std::begin(perDesktopData); data != std::end(perDesktopData);)
-                {
-                    if (data->deviceId == deviceId && data->zoneSetUuid == zoneSetId)
-                    {
-                        if (!IsAnotherWindowOfApplicationInstanceZoned(window, deviceId))
-                        {
-                            DWORD processId = 0;
-                            GetWindowThreadProcessId(window, &processId);
-
-                            data->processIdToHandleMap.erase(processId);
-                        }
-
-                        // if there is another instance of same application placed in the same zone don't erase history
-                        size_t windowZoneStamp = reinterpret_cast<size_t>(::GetProp(window, MULTI_ZONE_STAMP));
-                        for (auto placedWindow : data->processIdToHandleMap)
-                        {
-                            size_t placedWindowZoneStamp = reinterpret_cast<size_t>(::GetProp(placedWindow.second, MULTI_ZONE_STAMP));
-                            if (IsWindow(placedWindow.second) && (windowZoneStamp == placedWindowZoneStamp))
-                            {
-                                return false;
-                            }
-                        }
-
-                        data = perDesktopData.erase(data);
-                        if (perDesktopData.empty())
-                        {
-                            appZoneHistoryMap.erase(processPath);
-                        }
-                        SaveFancyZonesData();
-                        return true;
-                    }
-                    else
-                    {
-                        ++data;
-                    }
-                }
+                data.zoneIndexSet.push_back(static_cast<int>(value.GetNumber()));
             }
         }
-
-        return false;
-    }
-
-    bool FancyZonesData::SetAppLastZones(HWND window, const std::wstring& deviceId, const std::wstring& zoneSetId, const std::vector<int>& zoneIndexSet)
-    {
-        std::scoped_lock lock{ dataLock };
-
-        if (IsAnotherWindowOfApplicationInstanceZoned(window, deviceId))
-        {
-            return false;
-        }
-
-        auto processPath = get_process_path(window);
-        if (processPath.empty())
-        {
-            return false;
-        }
-
-        DWORD processId = 0;
-        GetWindowThreadProcessId(window, &processId);
-
-        auto history = appZoneHistoryMap.find(processPath);
-        if (history != std::end(appZoneHistoryMap))
+        else if (json.HasKey(L"zone-index"))
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.